src/main/java/com/khutircraftubackend/services/registration/RegistrationSellerService.java

```diff
@@ -0,0 +1,72 @@
+package com.khutircraftubackend.services.registration;
+
+import com.khutircraftubackend.dto.SellerDTO;
+import com.khutircraftubackend.mapper.SellerMapper;
+import com.khutircraftubackend.models.SellerEntity;
+import com.khutircraftubackend.repositories.SellerRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class RegistrationSellerService {
+
+    private SellerRepository sellerRepository;
+    private JavaMailSender mailSender;
+    private BCryptPasswordEncoder passwordEncoder;
+
+    @Value("${spring.mail.sellerName")
+    private String mailSellerName;
+
+    @Transactional
+    public void registerSeller(SellerDTO sellerDTO) {
+        String confirmationCode = UUID.randomUUID().toString();
+        SellerEntity seller = SellerMapper.INSTANCE.toSellerEntity(sellerDTO);
+
+        seller.setPassword(passwordEncoder.encode(sellerDTO.password()));
+        seller.setEnabled(false);
+        seller.setConfirmationCode(confirmationCode);
+
+        sellerRepository.save(seller);
+
+        sendConfirmationEmail(seller.getEmail(), confirmationCode);
+    }
+    @Async
+    public void sendConfirmationEmail(String email, String confirmationCode) {
+        SimpleMailMessage mailMessage = new SimpleMailMessage();
+        mailMessage.setTo(email);
+        mailMessage.setFrom(mailSellerName);
+        mailMessage.setSubject("Підтвердження облікового запису");
+        mailMessage.setText("Для підтвердження вашого облікового запису перейдіть за посиланням: "
+                + "http://your-app-domain.com/api/register/confirm?code=" + confirmationCode);
```

Comment:
this supposed to be a front-end URI , no need to put `/api/` in it.
the front-end in that URI might use `/api/register/....whatever...` though

Reply:
need to be discussed